docs/badges/configuration.rst

```diff
@@ -0,0 +1,241 @@
+
+- feature configuration
+- badges management
+- requirements setup
+- backends configuration (optional)
+
+
+Feature configuration
+---------------------
+
+Badges are under a feature flag (disabled by default):
+
+.. code-block:: sh
+
+    # credentials settings:
+    ENABLE_BADGES = false
+
+Also the feature has its configuration which may look like this:
+
+.. code-block:: sh
+
+    # credentials settings:
+    BADGES_CONFIG = {
+        "processing": {
+            "processor": "credentials.apps.badges.processing.BadgeProcessor",
+            "events": [
+                "org.openedx.learning.course.grade.passed.v1",
+                "org.openedx.learning.course.grade.failed.v1",
+            ],
+        },
+        "collecting": {
+            "collector": "credentials.apps.badges.collecting.BadgeCollector",
+            "event_bus_topic": "badges",
+            "events": [
+                "org.openedx.learning.badge.awarded.v1",
+                "org.openedx.learning.badge.revoked.v1",
+            ],
+        },
+        "distribution": {
+            "credly": {
+
+            }
+        }
+    }
+
+Configuration consists of 2 parts:
+
+- `processing`_ includes items related to incoming events processing.
+- `collecting`_ includes items related to completed badges collecting.
+
+Incoming Events configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The feature explicitly configures list of system events which are taken into account during processing.
+`Badge Processor`_ is auto-connected to listen to those event types.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["processing"]["events"] = [
+        "org.openedx.learning.course.grade.passed.v1",
+        "org.openedx.learning.course.grade.failed.v1",
+        ...
+    ]
+
+Not registered here event types are ignored.
+
+Outgoing Events configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Additionally, badges configuration lists event types which must be registered for external emission:
+
+.. code-block:: sh
+
+    BADGES_CONFIG["collecting"]["events"] = [
+        "org.openedx.learning.badge.awarded.v1",
+        "org.openedx.learning.badge.revoked.v1",
+    ]
+
+By default these event types are dynamically auto-registered to EventBus producer in addition to already present in `EVENT_BUS_PRODUCER_CONFIG`.
+
+Badge Processor
+~~~~~~~~~~~~~~~~
+
+For incoming external events processing configuration points to an importable `Badge Processor`_ class definition.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["processing"]["processor"] = "credentials.apps.badges.processing.BadgeProcessor"
+
+Badge Collector
+~~~~~~~~~~~~~~~~
+
+For completed badges configuration points to an importable `Badge Collector`_ class definition.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["collecting"]["collector"] = "credentials.apps.badges.collecting.BadgeCollector"
+
+
+Badges management
+-----------------
+
+Badges management includes the following life-cycle stages or maintenance activities:
```

Comment:
I believe it will be good to add here state diagram for displaying lifecycle, something like this may be modified, and used in the documentation

https://showme.redstarplugin.com/s/s:4SIKdM94

![image](https://github.com/raccoongang/credentials/assets/22370912/9ca043ed-3e42-4d9d-bb77-c864bd0afe4c)